### PR TITLE
Refactor web extension

### PR DESCRIPTION
**Pull request by Jyyjy**
This PR refactors the web extension, most of the changes simply reduce repeated code but there are a few actual changes.

1. Fixes a bug where extension options were not updated in tab events.
2. Changes browser log format to be more consistent with main package logs, and include a bit more information.
3. Adds 2 additional tab events, onAttached and onUpdated. There are other tab events that still aren't logged, but it's not immediately clear how to log them. 
4. Instruments the extension options page because why not.

### CODE REVIEW

File: build/UserALEWebExtension/optionsPage.html [hunk 1]
```diff
@@ -19,31 +19,30 @@
...
     <br/>
 
     <label>User:</label>
-    <input id="toolUser"/>
+    <input id="user"/>
     <br/>
 
     <label>Tool Name:</label>
-    <input id="toolName"/>
+    <input id="tool"/>
```
Comment (EandrewJones) [inline build/UserALEWebExtension/optionsPage.html:41]: Is there a reason this differs from the javascript package naming conventions?

File: build/UserALEWebExtension/optionsPage.html [hunk 2]
```diff
@@ -19,31 +19,30 @@
...
     <br/>
 
     <label>Tool Version:</label>
-    <input id="toolVersion"/>
+    <input id="version"/>
```
Comment (EandrewJones) [inline build/UserALEWebExtension/optionsPage.html:45]: Same as above.

File: src/UserALEWebExtension/optionsPage.html [hunk 3]
```diff
@@ -19,31 +19,30 @@
...
     <br/>
 
     <label>User:</label>
-    <input id="toolUser"/>
+    <input id="user"/>
     <br/>
 
     <label>Tool Name:</label>
-    <input id="toolName"/>
+    <input id="tool"/>
```
Comment (EandrewJones) [inline src/UserALEWebExtension/optionsPage.html:41]: Same as previous comment.

**Review by EandrewJones — CHANGES_REQUESTED** [review]
Very minor comments on the code itself. 

Only question: Do we have any tests for the plugin? If not, why not? If yes, please add some to, at minimum, verify the code fixes the bugs and ensure the refactor doesn't break existing behavior or introduce new bugs.

The code changes appear to be a very nice simplification. But I am always hesitant to refactor without any way to validate.

**Comment by Jyyjy** [comment]
> Very minor comments on the code itself. 
> 
> 
> 
> Only question: Do we have any tests for the plugin? If not, why not? If yes, please add some to, at minimum, verify the code fixes the bugs and ensure the refactor doesn't break existing behavior or introduce new bugs.
> 
> 
> 
> The code changes appear to be a very nice simplification. But I am always hesitant to refactor without any way to validate.

No plug-in tests exist. I think the plug-in has historically been an experimental tool only. That said it's already in my notes to write some plug-in tests.

**Comment by EandrewJones** [comment]
Alright. I'm hesitantly approving since this is a blocker.

File: build/UserALEWebExtension/optionsPage.html (hunk 1 above)
Comment (Jyyjy) [reply build/UserALEWebExtension/optionsPage.html:41]: I changed the html id's to match the data tag names defined here:
https://github.com/apache/flagon-useralejs/blob/aac8e6766972e27bb593e777f0c9ca5ec84598ab/src/getInitialSettings.js#L25

Originally I intended to add a layer of abstraction that defines the tag names and js property names for each setting and could be imported into getInitialSettings.js and options.js. But I wanted to limit the scope of this PR to the web extension.

**Pull request merged**